python/cudf/cudf/core/index.py

```diff
@@ -466,8 +487,36 @@ def __init__(self, start, stop=None, name=None):
             start, stop = 0, start
         self._start = int(start)
         self._stop = int(stop)
-        self.name = name
         self._cached_values = None
+        self._index = None
+        self._name = name
+
+    @property
+    def name(self):
+        return self._name
+
+    @name.setter
+    def name(self, value):
+        self._name = value
+
+    def as_column(self):
```

Comment:
It feels really awkward that this method is public that bypasses a memoized cache, but the memoized property is internal via `._values`.

Reply:
Would it be high effort to `.as_column` --> `._as_column`?

Or remove it entirely and just use `._values`

Reply:
OK let me try doing that

Reply:
Done